src/taxonomy/data/apiHooks.jsx

```diff
@@ -26,6 +26,26 @@ const useTaxonomyListData = (org) => (
   })
 );
 
+/**
+ * Builds the mutation to delete a taxonomy.
+ * @returns {Object} - An object with the mutation configuration.
+ */
+export const useDeleteTaxonomy = () => {
+  /**
+   * Deletes a taxonomy.
+   *
+   * @param {import("./types.mjs").DeleteTaxonomyParams} params - Parameters for deleting the taxonomy.
+   * @returns {Promise<void>} - A promise that resolves when the taxonomy is deleted.
+   */
+  const deleteTaxonomyWrap = async ({ pk }) => {
+    await deleteTaxonomy(pk);
+  };
+
+  return useMutation({
+    mutationFn: deleteTaxonomyWrap,
+  });
+};
```

Comment:
Is there a reason why we dont use the `deleteTaxonomy` api function directly inside the `useMutation` without wrapping it? Maybe something similar to this (It might need some adjustments, as I haven't tested this, but I had something similar working in my [PR](https://github.com/openedx/frontend-app-course-authoring/pull/704/files#diff-be91f7fb28c98bca08df36adfddf620d961c7aebf0fe600b1953eabfe2af19b4R124))

```suggestion
/**
 * Deletes a taxonomy. 
 */
export const useDeleteTaxonomy = () => (
  useMutation({
    mutationFn: ({ pk }) => deleteTaxonomy(pk),
  });
);
```

Reply:
@yusuf-musleh I did this changes, a little copy of your PR:

```
/**
 * Builds the mutation to delete a taxonomy.
 * @returns {Object} - An object with the mutation configuration.
 */
export const useDeleteTaxonomy = () => {
  const queryClient = useQueryClient()

  /**
   * @type {import("@tanstack/react-query").MutateFunction<any, any, {pk: number}>}
  */
  return useMutation({
    mutationFn: async ({ pk }) => await deleteTaxonomy(pk),
    onSettled: () => {
      queryClient.invalidateQueries({ queryKey: ['taxonomyList'] });
    },
  });
};
```

But I got this error after run `npm run types`:

```
src/taxonomy/data/apiHooks.jsx:40:26 - error TS2339: Property 'pk' does not exist on type 'void'.

40     mutationFn: async ({ pk }) => await deleteTaxonomy(pk),
```

How did you resolve this error in your PR?

Reply:
@ChrisChV I figured it out, you have to put the jsdoc comment right above the `mutationFn:` line, mine looks like this, (notice how it is _inside_ the `useMutation`):

```jsx
/**
 * Builds the mutation to update the tags applied to the content object
 * @param {string} contentId The id of the content object to update tags for
 * @param {number} taxonomyId The id of the taxonomy the tags belong to
 */
export const useContentTaxonomyTagsMutation = (contentId, taxonomyId) => {
  const queryClient = useQueryClient();

  return useMutation({
    /**
     * @type {import("@tanstack/react-query").MutateFunction<
     *   any,
     *   any,
     *   {
     *     tags: string[]
     *   }
     * >}
     */
    mutationFn: ({ tags }) => updateContentTaxonomyTags(contentId, taxonomyId, tags),
    onSettled: () => {
      queryClient.invalidateQueries({ queryKey: ['contentTaxonomyTags', contentId] });
    },
  });
};

```

Reply:
Thanks! It works